### PR TITLE
fix rshiny build when behind proxy

### LINKED ISSUE #627: R-shiny quickstarter failed to build
**Describe the bug**
The build of the R-shiny quickstarter is falling to build the image from the dockerfile available in GH from the master branch

**To Reproduce**
1. Create a new OD project
2. Add the R-shiny quickstarter
3. The Openshift build stage will fail

**Expected behavior**
That the Openshift build stage works

**Screenshots**
image

**Affected version (please complete the following information):**
 - OpenShift: 3.11
 - OpenDevStack master

**Log Output (ensure to remove any confidential information like tokens, project names, etc.**
```
RUN apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys E298A3A825C0D65DFD57CBB651716619E084DAB9
--
  | ---> Running in c4071caf6b0f
  |  
  | Warning: apt-key output should not be parsed (stdout is not a terminal)
  | Executing: /tmp/apt-key-gpghome.woFugLUDbB/gpg.1.sh --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys E298A3A825C0D65DFD57CBB651716619E084DAB9
  | gpg: keyserver receive failed: End of file
  | Removing intermediate container c4071caf6b0f
  | error: build error: The command '/bin/sh -c apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys E298A3A825C0D65DFD57CBB651716619E084DAB9' returned a non-zero code: 2

```

**Additional context**
The error seems to come from here: https://github.com/opendevstack/ods-quickstarters/blob/master/ds-rshiny/files/docker/Dockerfile#L24

@felipecruz91 @jssnull

### PR DESCRIPTION
**Pull request by braisvq1996**
Fixes #627

Tested with and without proxy variable

FYI @felipecruz91 @gerardcl

### CODE REVIEW

File: ds-rshiny/files/docker/Dockerfile [hunk 1]
```diff
@@ -21,8 +21,13 @@ RUN echo 'force-unsafe-io' > /etc/dpkg/dpkg.cfg.d/02apt-speedup && \
...
-    apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys E298A3A825C0D65DFD57CBB651716619E084DAB9 && \
-    echo 'deb https://cloud.r-project.org/bin/linux/ubuntu focal-cran40/' > /etc/apt/sources.list.d/cran.list && \
+    if [ -n "$HTTP_PROXY" ]; then \
+        apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --keyserver-options http-proxy=$HTTP_PROXY --recv-keys E298A3A825C0D65DFD57CBB651716619E084DAB9 && \
+        echo 'deb https://cloud.r-project.org/bin/linux/ubuntu focal-cran40/' > /etc/apt/sources.list.d/cran.list; \
+    else \
+        apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys E298A3A825C0D65DFD57CBB651716619E084DAB9 && \
+        echo 'deb https://cloud.r-project.org/bin/linux/ubuntu focal-cran40/' > /etc/apt/sources.list.d/cran.list; \
+    fi && \
     # Install dependencies
```
Comment (gerardcl) [inline ds-rshiny/files/docker/Dockerfile:31]:
```suggestion
    if [ -n "$HTTP_PROXY" ]; then \
        apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --keyserver-options http-proxy=$HTTP_PROXY --recv-keys E298A3A825C0D65DFD57CBB651716619E084DAB9 && \
    else \
        apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys E298A3A825C0D65DFD57CBB651716619E084DAB9 && \
    fi
    echo 'deb https://cloud.r-project.org/bin/linux/ubuntu focal-cran40/' > /etc/apt/sources.list.d/cran.list; && \
    # Install dependencies
```

**Review by gerardcl — CHANGES_REQUESTED** [review]
I would move as suggested the repeated line to outside the condition

File: ds-rshiny/files/docker/Dockerfile (hunk 1 above)
Comment (braisvq1996) [reply ds-rshiny/files/docker/Dockerfile]: Done in latest commit

**Review by gerardcl — APPROVED** [review]
LGTM! 🚀

**Pull request merged**